tinyagentos/routes/desktop_browser/copilot_agent_ws.py

```
@@ -0,0 +1,154 @@
+
+def _extract_host(msg: dict) -> str:
+    """Extract the host the agent claims to operate on. Falls back to
+    parsing the url field if no explicit host."""
+    host = msg.get("host")
+    if isinstance(host, str) and host:
+        return host
+    url = msg.get("url")
+    if isinstance(url, str) and url:
+        try:
+            return urlparse(url).hostname or ""
+        except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)